tests/unit/s2n_client_hello_test.c

```diff
@@ -177,6 +177,170 @@ int main(int argc, char **argv)
+
+        struct s2n_client_hello *client_hello = s2n_connection_get_client_hello(server_conn);
+
+        /* Verify s2n_connection_get_client_hello returns the handle to the s2n_client_hello on the connection */
+        EXPECT_EQUAL(client_hello, &server_conn->client_hello);
+
+        uint8_t* collected_client_hello = client_hello->raw_message.blob.data;
+        uint16_t collected_client_hello_len = client_hello->raw_message.blob.size;
+
+        /* Verify collected client hello message length */
+        EXPECT_EQUAL(collected_client_hello_len, sslv2_client_hello_len);
+
+        /* Verify the collected client hello matches what was sent */
+        EXPECT_SUCCESS(memcmp(collected_client_hello, sslv2_client_hello, sslv2_client_hello_len));
+
+        /* Verify s2n_client_hello_get_raw_message_length correct */
+        EXPECT_EQUAL(s2n_client_hello_get_raw_message_length(client_hello), sslv2_client_hello_len);
+
+        uint8_t expected_cs[] = {
+	    0x00, 0xc0, 0x24, 0x00, 0xc0, 0x28, 0x00, 0x00,
+	    0x3d, 0x00, 0xc0, 0x26, 0x00, 0xc0, 0x2a, 0x00,
+	    0x00, 0x6b, 0x00, 0x00, 0x6a, 0x00, 0xc0, 0x0a,
+	    0x07, 0x00, 0xc0, 0x00, 0xc0, 0x14, 0x00, 0x00,
+	    0x35, 0x00, 0xc0, 0x05, 0x00, 0xc0, 0x0f, 0x00,
+	    0x00, 0x39, 0x00, 0x00, 0x38, 0x00, 0xc0, 0x23,
+	    0x00, 0xc0, 0x27, 0x00, 0x00, 0x3c, 0x00, 0xc0,
+	    0x25, 0x00, 0xc0, 0x29, 0x00, 0x00, 0x67, 0x00,
+	    0x00, 0x40, 0x00, 0xc0, 0x09, 0x06, 0x00, 0x40,
+	    0x00, 0xc0, 0x13, 0x00, 0x00, 0x2f, 0x00, 0xc0,
+	    0x04, 0x01, 0x00, 0x80, 0x00, 0xc0, 0x0e, 0x00,
+	    0x00, 0x33, 0x00, 0x00, 0x32, 0x00, 0xc0, 0x2c,
+	    0x00, 0xc0, 0x2b, 0x00, 0xc0, 0x30, 0x00, 0x00,
+	    0x9d, 0x00, 0xc0, 0x2e, 0x00, 0xc0, 0x32, 0x00,
+	    0x00, 0x9f, 0x00, 0x00, 0xa3, 0x00, 0xc0, 0x2f,
+	    0x00, 0x00, 0x9c, 0x00, 0xc0, 0x2d, 0x00, 0xc0,
+	    0x31, 0x00, 0x00, 0x9e, 0x00, 0x00, 0xa2, 0x00,
+	    0x00, 0xff,
+        };
+
+        /* Verify collected cipher_suites size correct */
+        EXPECT_EQUAL(client_hello->cipher_suites.size, sizeof(expected_cs));
+
+        /* Verify collected cipher_suites correct */
+        EXPECT_SUCCESS(memcmp(client_hello->cipher_suites.data, expected_cs, sizeof(expected_cs)));
+
+        /* Verify s2n_client_hello_get_cipher_suites_length correct */
+        EXPECT_EQUAL(s2n_client_hello_get_cipher_suites_length(client_hello), sizeof(expected_cs));
+
+        /* Verify collected extensions size correct */
+        EXPECT_EQUAL(client_hello->extensions.size, 0);
+
+        /* Verify s2n_client_hello_get_extensions_length correct */
+        EXPECT_EQUAL(s2n_client_hello_get_extensions_length(client_hello), 0);
+
+        /* Free all handshake data */
+        EXPECT_SUCCESS(s2n_connection_free_handshake(server_conn));
+
+        /* Verify connection_wipe resized the s2n_client_hello.raw_message stuffer back to 0 */
+        EXPECT_NULL(client_hello->raw_message.blob.data);
+        EXPECT_EQUAL(client_hello->raw_message.blob.size, 0);
```

Comment:
I saw these checks for the raw_message are also below the wipe_connection check a bit lower (lines 325-327).
But I see it's now free_handshake's responsibility to clear the raw_message, as well as the parsed_extensions. 
We should fix the comment here (to 'verify _free_handshake_ resized' instead of 'verify _connection_wipe_ resized'), as well as move all data clearing checks (325 ->336) here, with removal of the duplicate check for the raw_message.

Also noticed the responsibility to clean these structures remained in connection_wipe as well, but not sure if there's any test that checks connection_wipe in the absence of a free_handshake call.

Reply:
i can update the comment.